common/geth/client.go

```diff
@@ -31,6 +32,7 @@ type EthClient struct {
 	NoSendTransactOpts *bind.TransactOpts
 	Contracts          map[gethcommon.Address]*bind.BoundContract
 	Logger             common.Logger
+	numConfirmations   int
 }
```

Comment:
Do you think this is this is generally a global setting or would it be something that would vary between different types of calls?

Reply:
I think this should be something each component sets globally and hidden from callers for now, although i can see there may be a need to override for different types of calls in the future if we have a lot more different types of calls.